js/interface.js

```diff
@@ -1,78 +1,176 @@
-var $folderContents = $('#folder-contents');
+// VAR SETUP //
+var $folderContents = $('.file-table-body');
+var $folderList;
+var $organizationAppList = $('.dropdown-menu-holder ul');
 var templates = {
   file: template('file'),
-  folder: template('folder')
+  folder: template('folder'),
+  folderItem: template('folder-item'),
+  organizations: template('organizations'),
+  apps: template('apps')
 };
+
 var currentFolderId;
 var currentFolders;
 var currentFiles;
+var counter;
+
+// CORE FUNCTIONS //
+// Get organizations and apps list for left side menu
+function getOrganizationsList() {
+  counter = 0;
+  Fliplet.Organizations.get().then(function (organizations) {
+    organizations.forEach(addOrganizations);
+    $folderList = $('.folder-list');
+  });
+}
+
+function getAppsList() {
+  Fliplet.Apps.get().then(function (apps) {
+    apps.forEach(addApps);
+    $folderList = $('.folder-list');
+  });
+}
+
+// Get folders and files depending on ID (Org, App, Folder) to add to the content area
+function getFolderContents(el) {
+  if (el.attr('data-type') == "app") {
+    var currentAppId = el.attr('data-app-id');
+  } else if (el.attr('data-type') == "organization") {
+    var currentOrgId = el.attr('data-org-id');
+  } else {
+    var currentFolderId = el.attr('data-id');
+  }
 
-function getFolderContents(folderId) {
-  currentFolderId = folderId;
   currentFolders = [];
   currentFiles = [];
   $folderContents.html('');
 
-  $('[data-upload-file]').toggle(!!folderId);
+  Fliplet.Media.Folders.get({
+    organizationId: currentOrgId,
+    appId: currentAppId,
+    folderId: currentFolderId
+  }).then(function (response) {
+    response.folders.forEach(function(i) {
```

Comment:
Which will be passed through. In my code, you are passing a function as reference to be used from the `forEach` function when iterating the loop. All the arguments will be passed to `addFolder`